app/code/Magento/QuoteGraphQl/Model/Resolver/DownloadableLinks.php

```diff
@@ -0,0 +1,175 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\QuoteGraphQl\Model\Resolver;
+
+use Magento\Catalog\Model\Product;
+use Magento\Downloadable\Helper\Data as DownloadableHelper;
+use Magento\Downloadable\Model\LinkFactory;
+use Magento\Downloadable\Model\SampleFactory;
+use Magento\Framework\Data\Collection;
+use Magento\Framework\GraphQl\Query\EnumLookup;
+use Magento\Framework\Exception\LocalizedException;
+use Magento\Framework\GraphQl\Config\Element\Field;
+use Magento\Framework\GraphQl\Query\ResolverInterface;
+use Magento\Framework\GraphQl\Schema\Type\ResolveInfo;
+use Magento\Quote\Model\Quote\Item as QuoteItem;
+
+/**
+ * @inheritdoc
+ */
+class DownloadableLinks implements ResolverInterface
+{
+    /**
+     * @var EnumLookup
+     */
+    private $enumLookup;
+
+    /**
+     * @var DownloadableHelper
+     */
+    private $downloadableHelper;
+
+    /**
+     * @var LinkFactory
+     */
+    private $linkFactory;
+
+    /**
+     * @var SampleFactory
+     */
+    private $sampleFactory;
+
+    /**
+     *
+     * @param EnumLookup $enumLookup
+     * @param DownloadableHelper $downloadableHelper
+     * @param LinkFactory $linkFactory
+     * @param SampleFactory $sampleFactory
+     */
+    public function __construct(
+        EnumLookup $enumLookup,
+        DownloadableHelper $downloadableHelper,
+        LinkFactory $linkFactory,
+        SampleFactory $sampleFactory
+    ) {
+        $this->enumLookup = $enumLookup;
+        $this->downloadableHelper = $downloadableHelper;
+        $this->linkFactory = $linkFactory;
+        $this->sampleFactory = $sampleFactory;
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public function resolve(
+        Field $field,
+        $context,
+        ResolveInfo $info,
+        array $value = null,
+        array $args = null
+    ) {
+        if (!isset($value['model'])) {
+            throw new LocalizedException(__('"model" value should be specified'));
+        }
+
+        /** @var QuoteItem $quoteItem */
+        $quoteItem = $value['model'];
+
+        /** @var Product $product */
+        $product = $quoteItem->getProduct();
+
+        $data = null;
+        if (in_array($product->getTypeId(), ['downloadable', 'virtual'])) {
```

Comment:
Should be `InputGraphQlException` if passed product with the wrong type

Reply:
**implemented**